File: Storage/Entities/Product.cs
namespace Storage.Entities;

public class Product
{
    public Guid Id { get; set; }
    public required string Name { get; set; }
    public required string Description { get; set; }
    public decimal Price { get; set; }
    public decimal? DiscountPrice { get; set; }
    public int QuantitySold { get; set; }
    public int? QuantityLimit { get; set; }
    public List<Image>? Images { get; set; }
    public DateTimeOffset CreatedAt { get; set; }
    public int TotalCommentsQuantity { get; set; }
    public int TotalRating { get; set; }
    public List<Label>? Labels { get; set; }
    public bool IsVisible { get; set; } = true;
}

File: Angular/client-app/src/app/models/product.ts
import { Imagee } from "./image";
import { Label } from "./label";

export interface Product {
    id: string
    name: string;
    description: string;
    price: number;
    discountPrice: number | null;
    quantitySold: number;
    quantityLimit: number;
    images : Imagee[] | null; 
    totalCommentsQuantity: number;
    totalRating: number;
    labels: Label[] | null;
    isVisible: boolean;
    quantity: number;
    _quantity: number;
    _mainImageUrl: string;
    createdAt: string;
    disabled: boolean;
}